XiEditor/Document.swift

```diff
@@ -77,10 +77,10 @@ class Document: NSDocument {
         windowController.window?.minSize = Document.minWinSize
 
         self.editViewController = windowController.contentViewController as? EditViewController
+        editViewController?.document = self
         if let config = (NSApplication.shared.delegate as? AppDelegate)?.configCache {
             editViewController?.configChanged(changes: config)
```

Comment:
I think the problem is that there is at least _some_ setup that the view controller doesn't do unless certain methods are called. For instance, the code that sets up the unified toolbar is all in the setter for `EditViewController.unifiedToolbar`. This means that if we don't explicitly set this value at some point it will never run.

What we probably want is for `redrawEverything` (which gets called when the view is first loaded, for instance) to also handle this initial setup. This means that `redrawEverything` should grab the `configCache` and set `unifiedTitlebar`. 

This is sort of how themes work; you'll notice that `EditViewController` doesn't have a copy of the theme, it just has a reference to the shared theme that's owned by the `AppDelegate`. When the theme changes, we just tell the view controller that this has happened, and it goes and checks its reference for the updated values.